core/commands/pin.go

```diff
@@ -115,12 +118,31 @@ var addPinCmd = &cmds.Command{
 				}
 
 				if pv := v.Value(); pv != 0 {
-					out <- &AddPinOutput{Progress: v.Value()}
+					select {
+					case out <- &AddPinOutput{Progress: v.Value()}:
+					case <-ctx.Done():
+						log.Error(ctx.Err())
```

Comment:
This will print on the daemon, pretty sure we don't want that

Reply:
Agree.  Just return the error.